godtoolsTests/App/Share/Common/FileCache/FileCacheTests.swift

```diff
@@ -0,0 +1,253 @@
+@testable import godtools
+import Foundation
+
+@Suite(.serialized)
+struct FileCacheTests {
+    
+    private static let testsFileCacheRootDirectory: String = "tests_file_cache"
+    private static let tempDirectoryName: String = "temp_directory"
+    
+    @Test("")
+    func returnsRootDirectoryUrl() async throws {
+        
+        let fileCache = Self.getTestsFileCache()
+        
+        switch fileCache.getRootDirectory() {
+        case .success(let url):
+            #expect(url.absoluteString.contains(Self.testsFileCacheRootDirectory))
+        case .failure(let error):
+            throw error
+        }
+    }
+    
+    @Test("")
+    func returnsDirectoryUrl() async throws {
+        
+        let fileCache = Self.getTestsFileCache()
+        let location = FileCacheLocation(relativeUrlString: Self.testsFileCacheRootDirectory)
+        
+        switch fileCache.getDirectory(location: location) {
+        case .success(let url):
+            #expect(url.absoluteString.contains(Self.testsFileCacheRootDirectory))
+        case .failure(let error):
+            throw error
+        }
+    }
+    
+    @Test("")
+    func isDirectory() async throws {
+        
+        let fileCache = Self.getTestsFileCache()
+        let rootDirectoryUrl: URL
+        
+        switch fileCache.getRootDirectory() {
+        case .success(let url):
+            rootDirectoryUrl = url
+        case .failure(let error):
+            throw error
+        }
+        
+        let newDiretoryUrl: URL = rootDirectoryUrl.appendingPathComponent(Self.tempDirectoryName)
+        
+        switch fileCache.createDirectoryIfNotExists(directoryUrl: newDiretoryUrl) {
+        case .success( _):
+            break
+        case .failure(let error):
+            throw error
+        }
+        
+        let isDirectory: Bool = fileCache.getIsDirectory(url: newDiretoryUrl)
+        
+        #expect(isDirectory == true)
+    }
+    
+    @Test("")
+    func isNotDirectory() async throws {
+        
+        let fileCache = Self.getTestsFileCache()
+        let rootDirectoryUrl: URL
+        
+        switch fileCache.getRootDirectory() {
+        case .success(let url):
+            rootDirectoryUrl = url
+        case .failure(let error):
+            throw error
+        }
+        
+        let newDiretoryUrl: URL = rootDirectoryUrl.appendingPathComponent(Self.tempDirectoryName)
+        
+        let isDirectory: Bool = fileCache.getIsDirectory(url: newDiretoryUrl)
+        
+        #expect(isDirectory == false)
+    }
+    
+    @Test("")
+    func isFile() async throws {
+        
+        let fileCache = Self.getTestsFileCache()
+        
+        switch fileCache.getRootDirectory() {
+        case .success( _):
+            break
+        case .failure(let error):
+            throw error
+        }
+                
+        let optionalFileData: Data? = "text data".data(using: .utf8)
+        let fileData: Data = try #require(optionalFileData)
```

Comment:
@rachaelblue this is also a neat way to unwrap an optional using require and fail the test if the value is null since we can't proceed any further.